deliver/lib/deliver/runner.rb

```diff
@@ -62,6 +66,19 @@ def upload_metadata
       UploadAssets.new.upload(options) # e.g. app icon
     end
 
+    def prepare_app_icons(options = {})
+      app_icon_path = options[:app_icon] if options[:app_icon]
+      app_icon_metadata = File.join(options[:metadata_path], "app_icon.png")
```

Comment:
in fact it is optional, but in `detect_values.rb`

`options[:metadata_path] ||= File.join(containing, 'metadata')` this one is called everytime so it shouldn't be nil, i guess.

i added a return if not set - just in case.